docs/icd.md

```diff
@@ -0,0 +1,195 @@
+# Interface Control Document (ICD) - Services
+
+<center>
+
+<img src="https://github.com/Arrow-air/tf-github/raw/main/src/templates/doc-banner-services.png" style="height:250px" />
+
+</center>
+
+## :telescope: Overview
+
+This document details the common software interfaces required by all services in the Arrow ecosystem.
+
+Each service may add additional interfaces.
+
+Attribute | Description
+--- | ---
+Status | Draft
+
+## :books: Related Documents
+
+Document | Description
+--- | ---
+:construction: Requirements & User Stories :construction: | Requirements and user stories
+## :globe_with_meridians: Public API - REST
+
+A Public API is typically consumed by client applications, such as a website or mobile app.
+
+Features of REST:
+- Implements common HTTPS Transfer Protocol Methods (GET, PUT, POST, DELETE)
+- Respond in JSON, XML, CSV, and other formats
+
+### :scroll: Files
+A `lib-common` repository will contain REST interfaces that will be used by every microservice.
+
+File | Description
+---- | -----
+`common-rest.json` | REST interface definition, in OpenAPI Specification (OAS)
+`common-rest.rs` | Code stubs autogenerated from the common_rest.json file.
+
+The `lib-common` crate will be published to a registry and added as a dependency by each service, so these interfaces can be incorporated.
+```toml
+# e.g. svc-scheduler Cargo.toml
+[dependencies]
+lib-common = 0.1.2
+```
+
+Individual services may define further REST endpoints:
+
+```tree
+.
+├── Cargo.lock
+├── Cargo.toml # [workspace] members = ['client', 'server']
+├── Makefile
+├── client/
+├── proto/
+└── server
+    ├── build.rs
+    ├── Cargo.toml
+    ├── src
+    │   ├── main.rs
+    │   ├── public_api.rs # REST Endpoints
+```
+
+### :space_invader: Authentication
+
+OAuth2 servers will be used to authorize and authenticate users.
```

Comment:
We might not need to host these OAuth2 servers ourselves if we use SSO and allow users to choose which external service they want to use to log in with (eg: Google / Facebook / some other service).
In addition, we can use web3 to allow users to log in with their crypto wallet.

Reply:
Added that as a possibility: https://github.com/Arrow-air/se-services/pull/3/commits/53bbe5a7abe6b62682e2b655bb951f3423d7beca